page_creation.rst

```diff
@@ -65,44 +60,154 @@ random) number and prints it. To do that, create a "Controller class" and a
+
+    # config/routes.yaml
+
+    # the "app_lucky_number" route name is not important yet
+    app_lucky_number:
+        path: /lucky/number
+        controller: App\Controller\LuckyController::number
+
+That's it! If you are using Symfony web server, try it out by going to:
 
     http://localhost:8000/lucky/number
 
 If you see a lucky number being printed back to you, congratulations! But before
 you run off to play the lottery, check out how this works. Remember the two steps
 to creating a page?
 
-#. *Create a route*: The ``@Route`` above ``numberAction()`` is the *route*: it
-   defines the URL pattern for this page. You'll learn more about :doc:`routing </routing>`
-   in its own section, including how to make *variable* URLs;
+#. *Create a route*: In ``config/routes.yaml``, the route defines the URL to your
+    page (``path``) and what ``controller`` to call. You'll learn more about :doc:`routing </routing>`
+    in its own section, including how to make *variable* URLs;
 
-#. *Create a controller*: The method below the route - ``numberAction()`` - is called
-   the *controller*. This is a function where *you* build the page and ultimately
+#. *Create a controller*: This is a function where *you* build the page and ultimately
    return a ``Response`` object. You'll learn more about :doc:`controllers </controller>`
    in their own section, including how to return JSON responses.
 
+Annotation Routes
+-----------------
+
+Instead of defining your route in YAML, Symfony also allows you to use *annotation*
+routes. First, install the annotations package:
+
+.. code-block:: terminal
+
+    $ composer require annotations
+
+Then, in ``config/routes.yaml``, remove the route we just created and uncomment
+the annotation route import at the bottom:
+
+.. code-block:: yaml
+
+    controllers:
+        resource: ../src/Controller/
+        type: annotation
+
+After this one-time setup, you can nowadd your route directly *above* the controller:
+
+.. code-block:: diff
+
+    // src/Controller/LuckyController.php
+    // ...
+
+    + use Symfony\Component\Routing\Annotation\Route;
+
+    + /**
+    +  * @Route("/lucky/number")
+    +  */
+    class LuckyController
+    {
+        public function number()
+        {
+            // this looks exactly the same
+        }
+    }
+
+That's it! The page - ``http://localhost:8000/lucky/number`` will work exactly
+like before! Annotations are the recommended way to configure routes.
+
+Auto-Installing Recipes with Symfony Flex
+-----------------------------------------
+
+You may not have noticed, but when you ran ``composer require annotations``, two
+special things happened, both thanks to a powerful Composer plugin called
+:doc:`Flex </setup/flex>`.
+
+First, ``annotations`` isn't a real package name: it's an *alias* (i.e. shortcut)
+that Flex resolves to ``sensio/framework-extra-bundle``.
+
+Second, after this package was downloaded, Flex executed a *recipe*, which automatically
```

Comment:
In this paragraph I'm missing a simple description of what a recipe is. If you agree, I propose something like: *"a set of automated instructions that tell Symfony how to integrate some external package"*. So this is how it'd look:

```
Second, after this package was downloaded, Flex executed a *recipe*, which is a
set of automated instructions that tell Symfony how to integrate some external
package. Flex recipes exist for many packages (not just bundles) and have the
ability to do a lot, like adding configuration files, creating directories,
updating ``.gitignore`` and adding new config to your ``.env`` file. Flex
*automates* the installation of packages so you can get back to coding.
```